src/check_executor.rs

```diff
@@ -103,55 +115,78 @@ impl CheckResult {
 
 pub fn run_executor(
     concurrency: usize,
+    failure_retries: u16,
     checker: Arc<impl Checker + 'static>,
     producer: Arc<impl ResultsProducer + 'static>,
     region: String,
-) -> (CheckSender, JoinHandle<()>) {
+) -> (Arc<CheckSender>, JoinHandle<()>) {
     tracing::info!("executor.starting");
 
-    let (check_sender, receiver) = CheckSender::new();
-    let queue_size = check_sender.queue_size.clone();
-    let num_running = check_sender.num_running.clone();
+    let (sender, reciever) = CheckSender::new();
+    let queue_size = sender.queue_size.clone();
+    let num_running = sender.num_running.clone();
+
+    let check_sender = Arc::new(sender);
```

Comment:
the sender now needs to be owned by the executor AND the external service sending checks, so we put it into an arc.